core/src/main/java/JwtArrayUtils.java

```diff
@@ -0,0 +1,44 @@
+package pdi.jwt;
+
+public final class JwtArrayUtils {
+
+    private JwtArrayUtils() {}
+
+    /**
+     * A constant time equals comparison - does not terminate early if
+     * test will fail. For best results always pass the expected value
+     * as the first parameter.
+     *
+     * Ported from BouncyCastle to remove the need for a runtime dependency.
+     *
+     * @param expected first array
+     * @param supplied second array
+     * @return true if arrays equal, false otherwise.
+     */
+    public static boolean constantTimeAreEqual(byte[] expected, byte[] supplied)
+    {
+        if (expected == supplied)
+        {
+            return true;
+        }
+
+        if (expected == null || supplied == null)
+        {
+            return false;
+        }
+
+        if (expected.length != supplied.length)
+        {
+            return !JwtArrayUtils.constantTimeAreEqual(expected, expected);
```

Comment:
Could you add a link to the original source code in the description? I would like to take a look. This line seems strange. Since `expected == expected`, it will most likely return `true` immediately from L22 and then negate it. Would be faster to just return `false` here.

Reply:
Sure, not a problem. You can find the link here: https://github.com/bcgit/bc-java/blob/290df7b4edfc77b32d55d0a329bf15ef5b98733b/core/src/main/java/org/bouncycastle/util/Arrays.java#L145. FWIW, it looks like this snippet originates from an older commit and there has been some recent development in that area to improve the function. If you'd prefer a particular revision, I'd be happy to use that over this revision.